delta/sdk/src/main/scala/ch/epfl/bluebrain/nexus/delta/sdk/identities/IdentitiesImpl.scala

```diff
@@ -61,6 +62,14 @@ class IdentitiesImpl private[identities] (
       )
     }
 
+    def fetchRealm(parsedToken: ParsedToken): IO[Realm] =
+      realm
+        .getOrElseUpdate(parsedToken.rawToken, findActiveRealm(parsedToken.issuer))
+        .flatMap {
+          case Some(realm) => IO.pure(realm)
+          case None        => IO.raiseError(UnknownAccessTokenIssuer)
+        }
```

Comment:
Stylistic nit (not sure if we use monad transformers but they can be useful tools for isolated things like this)
```suggestion
      OptionT(realm.getOrElseUpdate(parsedToken.rawToken, findActiveRealm(parsedToken.issuer))).getOrRaise(UnknownAccessTokenIssuer)
```

Reply:
I did a variation of that